packages/hooks/src/use-field-id/use-field-id.ts

```diff
@@ -13,7 +14,7 @@ const useFieldId = (id: string, createIdFn: CreateIdFn): string => {
     setId(getFieldId(props, state, createIdFn));
   }, [id, internalId, setId, createIdFn]);
 
-  return internalId;
+  return internalId as string;
```

Comment:
we have to cast this, however at this point we should rely on the `internalId` being finally resolved via the `createIdFn` which is a required param